internal/pkg/mapper/configmetadata/remote.go

```diff
@@ -0,0 +1,86 @@
+package configmetadata
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/remote"
+	"github.com/keboola/keboola-as-code/internal/pkg/state"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+// configMetadataMapper add metadata to configurations loaded from API.
+type configMetadataMapper struct {
+	dependencies
+	logger         log.Logger
+	configMetadata map[model.BranchId]remote.ConfigMetadataResponse
+}
+
+type dependencies interface {
+	StorageApi() (*remote.StorageApi, error)
+}
+
+func NewMapper(s *state.State, d dependencies) *configMetadataMapper {
+	return &configMetadataMapper{dependencies: d, logger: s.Logger(), configMetadata: make(map[model.BranchId]remote.ConfigMetadataResponse)}
+}
+
+func (m *configMetadataMapper) OnRemoteChange(changes *model.RemoteChanges) error {
+	// Process loaded objects
+	errors := utils.NewMultiError()
+	for _, objectState := range changes.Loaded() {
+		if err := m.onRemoteLoad(objectState); err != nil {
```

Comment:
Tu by potom bola uz iba podmienka 
`if config, ok := objectState.RemoteState().(*model.Config), ...` 
a `if metadata, ok := metadataMap[config.Key().String()]; ok`

alebo by si to nechal, tak ako to je a do `onRemoteLoad` by si poslal `metadataMap`.